docs/reference/esql/time-spans.asciidoc

```diff
@@ -0,0 +1,106 @@
+[[esql-time-spans]]
+=== {esql} time spans
+
+++++
+<titleabbrev>Time spans</titleabbrev>
+++++
+
+Time spans provide a way of describing a specific period of time that separates two datetime values. There are currently two supported types of time spans, `DATE_PERIOD` specifying intervals in years, quarters, months, weeks and days, and `TIME_DURATION` specifying intervals in hours, minutes, seconds and milliseconds.
+
+The time span contains two elements, an integer value and a temporal unit, both are required when specifying a time span.
+
+Time spans are often used in datetime related grouping functions like <<esql-bucket, BUCKET>>, scalar functions like <<esql-date_trunc, DATE_TRUNC>> and arithmetic operators like <<esql-add, `+`>> and <<esql-subtract, `-`>>. There are conversions functions that can convert string literals to time spans, <<esql-to_dateperiod, TO_DATEPERIOD>>, <<esql-to_timeduration, TO_TIMEDURATION>> and their corresponding <<esql-cast-operator, cast operators>> `::DATE_PERIOD` and `::TIME_DURATION`.
+
+==== Examples of using time spans in {esql}
```

Comment:
```suggestion
[discrete]
[[esql-time-spans]]
==== Examples of using time spans in {esql}
```

Reply:
Do you mean `[[esql-time-spans-examples]]` here? `[[esql-time-spans]]` has already been used at the beginning of this file.

Reply:
certainly did! fixed :)